File: leetcode/136-single-number/single-number.java
class Solution {
    public int singleNumber(int[] nums) {
        HashMap<Integer,Integer> num=new HashMap<>();
        for(int i:nums){
            num.put(i,num.getOrDefault(i,0)+1);
        }
        for(int i:num.keySet()){
            if(num.get(i)==1){
                return i;
            }
        }
        return 0;
    }
}

File: leetcode/136-single-number/single-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
    let freq = new Map();

    // Count frequency
    for (let num of nums) {
        freq.set(num, (freq.get(num) || 0) + 1);
    }

    // Find the number with frequency 1
    for (let [key, value] of freq) {
        if (value === 1) {
            return key;
        }
    }
};